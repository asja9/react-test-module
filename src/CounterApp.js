//Agregando hook use
import React, { useState } from "react";
import PropTypes from 'prop-types';


const CounterApp = ( {value = 100}) =>{


    /**Eventos */
    /*const handleAdd = (event) =>{
         console.log(event);
    }*/

    //UseState-Hook
    /**[variable, funcion] */
    const[counter, setCounter] = useState(0);
    const  handleAdd = () =>{
        setCounter(counter+1);
        //da error counter++, intente cambiar el valor de una constante
        //usando una funcion
        //setCounter( (c) => c + 1 );
    }

    const  handleRemove = () =>{
        setCounter(counter-1);
        //da error counter++, intente cambiar el valor de una constante
        //usando una funcion
        //setCounter( (c) => c + 1 );
    }

    const  handleReset = () =>{
        //usando una funcion
        setCounter( (c) => c = 0 );
    }



    /*
    return (
        <>
         <h1>CounterApp</h1>
         <h2>{value}</h2>
         <button className="btn btn-dark btn-lg" onClick={ (e) => { handleAdd(e) }}>Agrega 1</button>
        </>
      );*/

      return (
        <>
         <h1>CounterApp</h1>
         <h2 className="ct">{value}</h2>
         <h2 id="counter">{counter}</h2>
         <button className="btn btn-dark btn-lg" id="sumar" onClick={ handleAdd }>Agrega 1</button>
         <button className="btn btn-dark btn-lg" id="restar" onClick={ handleRemove }>Remover 1</button>
         <button className="btn btn-dark btn-lg" onClick={ handleReset }>Reset</button>

        </>
      );

}

CounterApp.propTypes = {
    value: PropTypes.number
}



export default CounterApp;

//creacion rapida de la estructura rafcp