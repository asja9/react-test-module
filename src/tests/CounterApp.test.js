import CounterApp from '../CounterApp';
import React from 'react';
import { shallow } from 'enzyme';


describe('Pruebas en <CounterApp> usando Enzyme', () =>{

    let wrapper = shallow(<CounterApp />);
    // Se usa en todas las pruebas y posiblemente requiero reiniciar el valor
    // dentro de cada test

    //Reinicio
    beforeEach( () => {
       wrapper = shallow(<CounterApp />);
    });




    test("Debe de mostrar <CounterApp> correctamente, hacer match con un snapshot", () =>{  
      
        const counter = 100;
        //Shallow se encarga del render HTML realmente me devuleve y contiene todo
        //el HTML de la funcion que se prueba.
        const wrapper = shallow(<CounterApp contador = {counter} />);
        //Prueba de instantaneas, compara con un archivo de instantanea de referencia
        //almacenada junto a la prueba- ver carpeta snapshots
        expect(wrapper).toMatchSnapshot()
    });  


    //Verificando el defaukl 
    test("Debe de mostrar <CounterApp> el valor por defecto 100", () =>{  
        
        //Se espera que de devuelva un 100
        const valueDefault = '100';
        expect(wrapper).toMatchSnapshot();

        //Trim borra los espacios
        const counterFromModule = wrapper.find('.ct').text().trim();
        expect(counterFromModule).toBe(valueDefault);
    });  

    /* Simulacion de clics */
    test('Debe de icrementar el contador en 3', () =>{


        //Saber el valor antes de simular el clic
        //console.log(wrapper.find('#counter').text()); 
        //Dentro de html que se me devuelve voy a bucar el primer botton
        //Cada linea simula un clic
        wrapper.find('#sumar').simulate('click'); 
        wrapper.find('#sumar').simulate('click'); 
        wrapper.find('#sumar').simulate('click'); 

        const counter = wrapper.find('#counter').text().trim();
        expect(counter).toBe('3');        

    });
    

    test('Debe de decrementar el contador en -3', () =>{

        //Saber el valor antes de simular el clic
        console.log(wrapper.find('#counter').text()); 
        //Dentro de html que se me devuelve voy a bucar el primer botton
        //Cada linea simula un clic
        wrapper.find('#restar').simulate('click'); 
        wrapper.find('#restar').simulate('click'); 
        wrapper.find('#restar').simulate('click'); 
        wrapper.find('#restar').simulate('click'); 

        console.log(wrapper.find('#counter').text()); 

        const counter = wrapper.find('#counter').text().trim();
        expect(counter).toBe('-4');         

    });  

});


    