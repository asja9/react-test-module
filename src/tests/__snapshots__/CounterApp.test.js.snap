// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp> usando Enzyme Debe de mostrar <CounterApp> correctamente, hacer match con un snapshot 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2
    className="ct"
  >
    100
  </h2>
  <h2
    id="counter"
  >
    0
  </h2>
  <button
    className="btn btn-dark btn-lg"
    id="sumar"
    onClick={[Function]}
  >
    Agrega 1
  </button>
  <button
    className="btn btn-dark btn-lg"
    id="restar"
    onClick={[Function]}
  >
    Remover 1
  </button>
  <button
    className="btn btn-dark btn-lg"
    onClick={[Function]}
  >
    Reset
  </button>
</Fragment>
`;

exports[`Pruebas en <CounterApp> usando Enzyme Debe de mostrar <CounterApp> el valor por defecto 100 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2
    className="ct"
  >
    100
  </h2>
  <h2
    id="counter"
  >
    0
  </h2>
  <button
    className="btn btn-dark btn-lg"
    id="sumar"
    onClick={[Function]}
  >
    Agrega 1
  </button>
  <button
    className="btn btn-dark btn-lg"
    id="restar"
    onClick={[Function]}
  >
    Remover 1
  </button>
  <button
    className="btn btn-dark btn-lg"
    onClick={[Function]}
  >
    Reset
  </button>
</Fragment>
`;
